File: app/src/main/java/com/krystianwsul/checkme/domainmodel/local/LocalScheduleBridge.java
package com.krystianwsul.checkme.domainmodel.local;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.util.Pair;

import com.krystianwsul.checkme.domainmodel.ScheduleBridge;
import com.krystianwsul.checkme.persistencemodel.ScheduleRecord;
import com.krystianwsul.checkme.utils.TaskKey;

abstract class LocalScheduleBridge implements ScheduleBridge {
    @NonNull
    final ScheduleRecord mScheduleRecord;

    LocalScheduleBridge(@NonNull ScheduleRecord scheduleRecord) {
        mScheduleRecord = scheduleRecord;
    }

    @Override
    public long getStartTime() {
        return mScheduleRecord.getStartTime();
    }

    @Nullable
    @Override
    public Long getEndTime() {
        return mScheduleRecord.getEndTime();
    }

    @Override
    public void setEndTime(long endTime) {
        mScheduleRecord.setEndTime(endTime);
    }

    @NonNull
    @Override
    public TaskKey getRootTaskKey() {
        return new TaskKey(mScheduleRecord.getRootTaskId());
    }

    @Nullable
    @Override
    public Pair<String, String> getRemoteCustomTimeKey() {
        return null;
    }
}

File: app/src/main/java/com/krystianwsul/checkme/domainmodel/local/LocalScheduleBridge.kt
package com.krystianwsul.checkme.domainmodel.local

import com.krystianwsul.checkme.domainmodel.ScheduleBridge
import com.krystianwsul.checkme.persistencemodel.ScheduleRecord
import com.krystianwsul.checkme.utils.RemoteCustomTimeId
import com.krystianwsul.checkme.utils.ScheduleId
import com.krystianwsul.checkme.utils.TaskKey

abstract class LocalScheduleBridge(val scheduleRecord: ScheduleRecord) : ScheduleBridge {

    override val startTime by lazy { scheduleRecord.startTime }

    override val rootTaskKey get() = TaskKey(scheduleRecord.rootTaskId)

    override val remoteCustomTimeKey: Pair<String, RemoteCustomTimeId>? = null

    override var endTime
        get() = scheduleRecord.endTime
        set(value) {
            scheduleRecord.endTime = value
        }

    override val scheduleId by lazy { ScheduleId.Local(scheduleRecord.id) }
}
